config/sanitizer/tools.cmake

```diff
@@ -32,6 +41,28 @@ macro(cppcheck)
   endif()
 endmacro()
 
+find_program(CLANG_TIDY_EXE NAMES "clang-tidy")
+mark_as_advanced(FORCE CLANG_TIDY_EXE)
+if(CLANG_TIDY_EXE)
+  message(STATUS "clang-tidy found: ${CLANG_TIDY_EXE}")
+  if(NOT CLANG_TIDY)
+    message(STATUS "clang-tidy NOT ENABLED via 'CLANG_TIDY' variable!")
+    set(CMAKE_CXX_CLANG_TIDY
+        ""
+        CACHE STRING "" FORCE) # delete it
+  endif()
+elseif(CLANG_TIDY)
+  message(SEND_ERROR "Cannot enable clang-tidy, as executable not found!")
+  set(CMAKE_CXX_CLANG_TIDY
+      ""
+      CACHE STRING "" FORCE) # delete it
+else()
+  message(STATUS "clang-tidy not found!")
+  set(CMAKE_CXX_CLANG_TIDY
+      ""
+      CACHE STRING "" FORCE) # delete it
+endif()
```

Comment:
IIRC, clang-tidy is C++-specific and thus doesn't belong here

Reply:
will remove - got a little carried away with compare-merge.